src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs

```diff
@@ -254,7 +254,7 @@ private void WriteRequestBuilderBody(CodeClass parentClass, CodeMethod codeEleme
     private static void AddNullChecks(CodeMethod codeElement, LanguageWriter writer)
     {
         if (!codeElement.IsOverload)
-            foreach (var parameter in codeElement.Parameters.Where(static x => !x.Optional).OrderBy(static x => x.Name))
+            foreach (var parameter in codeElement.Parameters.Where(static x => !x.Optional && !x.IsOfKind(CodeParameterKind.RequestAdapter, CodeParameterKind.PathParameters)).OrderBy(static x => x.Name, StringComparer.OrdinalIgnoreCase))
```

Comment:
Do we also need to check for  `CodeParameterKind.RawUrl`?

Reply:
this method is only called when not in the raw url contructor overload case.